lib/fs.js

```diff
@@ -2070,11 +2069,52 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     } else {
       offset += written;
       length -= written;
-      writeAll(fd, isUserFd, buffer, offset, length, signal, callback);
+      writeAll(
+        fd, isUserFd, buffer, offset, length, signal, encoding, callback);
     }
   });
 }
 
+async function writeAllCustomIterable(
+  fd, isUserFd, buffer, offset, length, signal, encoding, callback) {
+  if (signal?.aborted) {
+    handleWriteAllErrorCallback(fd, isUserFd, new AbortError(), callback);
+    return;
+  }
+
+  const result = await buffer.next();
+  if (result.done) {
+    if (isUserFd) {
+      callback(null);
+    } else {
+      fs.close(fd, callback);
+    }
+    return;
+  }
+  const resultValue = isArrayBufferView(result.value) ?
```

Comment:
It makes sense to add an additional abort signal check before this line (as there's an await a few lines above)